### DISCUSSION TITLE
`not` in pipes

### DISCUSSION THREAD

**Discussion by vtu-dog** [nushell/nushell#11217] [Q&A] [answered]
Is there a way to negate output when using pipes?

Here's an example function:

```nushell
# Calls `my-command` if present.
def my-fun [
    ...args: string  # The arguments to pass to `my-command`.
] {
    if not (which my-command | is-empty) {
        ^my-command ($args | str join " ")
    } else {
        echo "not found"
    }
}
```

While this works, it annoys me to no end that I can't just write:

```nushell
# snip
if (which my-command | is-empty | not) {
# snip
```

I understand that this might be a limitation of how the built-in `not` works. In order to circumvent that behaviour, I tried writing my own `negate` function which I could use like this:

```nushell
# snip
if (which my-command | is-empty | negate) {
# snip
```

I can't get such a function right though. Any ideas?

**Comment by amtoine** [comment, marked as answer]
you can write
```nushell
not (pipeline)
```
as you said or
```nushell
pipeline | not $in
```

**Reply by vtu-dog** [reply]
Thanks! Do you know of any other implicit variables? I did not realise something like `$in` existed until now.

**Reply by amtoine** [reply]
happy to help :blush:

> Do you know of any other implicit variables?
>
i think it's the only one, afaik :yum: 
there is also `$it` in `where` but it's just a shortcut for `where {|it| $it.name}` where you can directly write `where $it.name == "foo"`

> I did not realise something like `$in` existed until now.
>
it should be easier to discover in the docs :cry:
